src/containerapp/azext_containerapp/custom.py

```diff
@@ -1510,6 +1519,963 @@ def delete_managed_environment(cmd, name, resource_group_name, no_wait=False):
+            r = _ensure_identity_resource_id(subscription_id, resource_group_name, r)
+            identity_def["userAssignedIdentities"][r] = {}  # pylint: disable=unsupported-assignment-operation
+
+    resources_def = None
+    if cpu is not None or memory is not None:
+        resources_def = ContainerResourcesModel
+        resources_def["cpu"] = cpu
+        resources_def["memory"] = memory
+
+    container_def = ContainerModel
+    container_def["name"] = container_name if container_name else name
+    container_def["image"] = image if not is_registry_msi_system(registry_identity) else HELLO_WORLD_IMAGE
+    if env_vars is not None:
+        container_def["env"] = parse_env_var_flags(env_vars)
+    if startup_command is not None:
+        container_def["command"] = startup_command
+    if args is not None:
+        container_def["args"] = args
+    if resources_def is not None:
+        container_def["resources"] = resources_def
+
+    template_def = JobTemplateModel
+    template_def["containers"] = [container_def]
+
+    containerappjob_def = ContainerAppsJobModel
+    containerappjob_def["location"] = location
+    containerappjob_def["identity"] = identity_def
+    containerappjob_def["properties"]["environmentId"] = managed_env
+    containerappjob_def["properties"]["configuration"] = config_def
+    containerappjob_def["properties"]["template"] = template_def
+    containerappjob_def["tags"] = tags
+
+    if workload_profile_name:
+        containerappjob_def["properties"]["workloadProfileName"] = workload_profile_name
+        ensure_workload_profile_supported(cmd, managed_env_name, managed_env_rg, workload_profile_name, managed_env_info)
+
+    if registry_identity:
+        if is_registry_msi_system(registry_identity):
+            set_managed_identity(cmd, resource_group_name, containerappjob_def, system_assigned=True)
+        else:
+            set_managed_identity(cmd, resource_group_name, containerappjob_def, user_assigned=[registry_identity])
+    try:
+        r = ContainerAppsJobClient.create_or_update(
+            cmd=cmd, resource_group_name=resource_group_name, name=name, containerapp_job_envelope=containerappjob_def, no_wait=no_wait)
+
+        if is_registry_msi_system(registry_identity):
+            while r["properties"]["provisioningState"] == "InProgress":
+                r = ContainerAppClient.show(cmd, resource_group_name, name)
+                time.sleep(10)
+            logger.info("Creating an acrpull role assignment for the system identity")
+            system_sp = r["identity"]["principalId"]
+            create_acrpull_role_assignment(cmd, registry_server, registry_identity=None, service_principal=system_sp)
+            container_def["image"] = image
+
+            registries_def = RegistryCredentialsModel
+            registries_def["server"] = registry_server
+            registries_def["identity"] = registry_identity
+            config_def["registries"] = [registries_def]
```

Comment:
where config_def is used for the following operation?

Reply:
similar implementation in Container Apps, we might need to fix both.. will update in next PR